src/libraries/Common/src/Roslyn/MetadataHelpers.cs

```diff
@@ -0,0 +1,1028 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Text;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class MetadataHelpers
+    {
+        public const char DotDelimiter = '.';
+        public const string DotDelimiterString = ".";
+        public const char GenericTypeNameManglingChar = '`';
+#if false
+        private const string GenericTypeNameManglingString = "`";
+#endif
+        public const int MaxStringLengthForParamSize = 22;
+        public const int MaxStringLengthForIntToStringConversion = 22;
+        public const string SystemString = "System";
+
+        // These can appear in the interface name that precedes an explicit interface implementation member.
+        public const char MangledNameRegionStartChar = '<';
+        public const char MangledNameRegionEndChar = '>';
+
+#if false
```

Comment:
this type is only minimally used.  it is mostly ifdef'ed out except for the only apis we use elsewhere.

Reply:
will do!